### PR TITLE
Fix e2e and autoconnect

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Fix issue with e2e not being able to select a token:
image

Also solves https://github.com/gnosis/cowswap/issues/1830

This is what happens:
image

The issue is related to https://github.com/bvaughn/react-window that is used for the selector
This component phiscally remove elements that are not visible from the DOM while you scroll. So obviously Cypress can find it

See more context
https://gnosisinc.slack.com/archives/C025G521XQD/p1637143944007100

## Solution

image

Basically the solution was to use the URL to select the tokens, it's easier :) 

Also, there was another error. The auto-connect logic implemented by @nenadV91 broke the automatic connections for e2e. The solution was quite easy, but pls Nenad double check what i did there i case u see any issue. 

## Not included
I left a TODO for an improvement. Also mentioned it in the thread in case @MareenG can implement it. Basically would 

# Test
1. See tests are passing

### CODE REVIEW

File: src/custom/hooks/web3.ts [hunk 1]
```diff
@@ -30,11 +30,12 @@ export function useEagerConnect() {
 
       // if there is no last saved provider set tried state to true
       if (!latestProvider) {
-        setTried(true)
-      }
-
-      // check if the last saved provider is Metamask
-      if (latestProvider === WalletProvider.INJECTED) {
+        // Try to auto-connect to the injected wallet
+        activate(injected, undefined, true).catch(() => {
```
Comment (nenadV91) [inline src/custom/hooks/web3.ts:34]: Just out of curiosity, is there a reason why we would try to connect to injected if there is no latestProvider?

**Review by nenadV91 — APPROVED** [review]

File: src/custom/hooks/web3.ts (hunk 1 above)
Comment (anxolin) [reply src/custom/hooks/web3.ts:34]: i answer u in our call, but now u see this fix https://github.com/gnosis/cowswap/issues/1830 then u get a stronger answer :)

**Review by alfetopito — APPROVED** [review]

**Comment by elena-zh** [comment]
I retested connection issue: works great now!
